nilearn/signal.py

```diff
@@ -280,8 +280,16 @@ def _check_wn(btype, freq, nyq):
 
 
 @fill_doc
-def butterworth(signals, sampling_rate, low_pass=None, high_pass=None,
-                order=5, copy=False):
+def butterworth(
+    signals,
+    sampling_rate,
+    low_pass=None,
+    high_pass=None,
+    order=5,
+    padtype="odd",
+    padlen=None,
```

Comment:
These are the default values for `scipy.signal.filtfilt`, so behavior won't change. I'm not sure if any of them work particularly well for BOLD data though.

Reply:
Ok I suppose we can keep it that way for this PR but can consider optimizing them in a separate one

Reply:
Agreed. I'm not sure how to actually do that though. If there's no principled way to decide between different padding approaches, then I think we'll have to rely on empirical testing across a range of datasets, which is a whole lot of work.

Reply:
My intuition is that this will have limited impact on final computations (beta maps, correlation, statistical tests). WDYT ?

Reply:
I agree that it's probably not going to have a large, or consistent, effect on results. When I tried to search the literature for mentions of padding, almost all of them mentioned zero-padding. Scipy doesn't implement zero-padding, but I think constant-padding is pretty similar, conceptually speaking. So maybe we could default to that, based on the fact that it's similar to what some previous fMRI papers have done?

Reply:
I have no better suggestion.

Reply:
Do you mind if I change the default in a separate PR? I'd like there to be a separation of concerns (plus the change to the default is kind of a breaking change).

Reply:
I agree for doing that in a separate PR